node/bft/ledger-service/src/lib.rs

```diff
@@ -60,3 +67,38 @@ macro_rules! spawn_blocking {
         }
     };
 }
+
+#[cfg(feature = "metrics")]
+fn update_block_metrics<N: Network>(block: &Block<N>) {
```

Comment:
My personal opinion on where this should go is in the `metrics` crate. This way if there's any question about the metrics collection logic, we have a singular place to look. Does anyone have any other opinions on where this should live?

Reply:
Pending other reviewer's comments on where this function should live, this has my approval.